libraries/helpers.rb

```diff
@@ -96,24 +96,24 @@ def product_lookup(product, version = '0.0.0')
       # implying latest, we need to additionally ensure that the
       # bottom version is something valid. If we don't have the check
       # in the elsif, it will say that 0.0.0 is not a valid version.
-      if (product == 'chef-server')
+      if product == 'chef-server'
         if (v < Mixlib::Versioning.parse('12.0.0')) && (v > Mixlib::Versioning.parse('11.0.0'))
           data['package-name'] = 'chef-server'
-        elsif (v < Mixlib::Versioning.parse('11.0.0')) && (v > Mixlib::Versioning.parse('1.0.0'))
+        elsif (v < Mixlib::Versioning.parse('11.0.0')) && (v > Mixlib::Versioning.parse('0.0.0'))
           Chef::Log.fatal("Invalid version specified, '#{version}' for #{product}!")
           fail
         end
-      elsif (product == 'manage') && (v < Mixlib::Versioning.parse('2.0.0'))
+      elsif (product == 'manage') && ((v < Mixlib::Versioning.parse('2.0.0')) && (v > Mixlib::Versioning.parse('0.0.0')))
         data['package-name'] = 'opscode-manage'
         data['ctl-command'] = 'opscode-manage-ctl'
       # TODO: When Chef Push server and client 2.0 are released, we
       # need to implement similar logic to chef-server, so that the
       # default "latest" version, 0.0.0 (no constraint) doesn't result
       # in the old package.
-      elsif (product == 'push-server') && (v < Mixlib::Versioning.parse('2.0.0'))
+      elsif (product == 'push-server') && ((v < Mixlib::Versioning.parse('2.0.0')) && (v > Mixlib::Versioning.parse('0.0.0')))
```

Comment:
I don't have a ton of context around this, but it seems worth noting that when I run this via delivery-cluster on ubuntu 14.04, the manage install fails with this version check. (Removing it fixes it.) Should we remove this check for now?